configure

```diff
@@ -769,8 +790,8 @@ case "${ARCH}" in
 
         case "${ARCH}" in
             x86_64)
-                CFLAGS="${CFLAGS} -DX86_64 -DX86_NOCHECK_SSE2"
-                SFLAGS="${SFLAGS} -DX86_64 -DX86_NOCHECK_SSE2"
+                CFLAGS="${CFLAGS} -DX86_NOCHECK_SSE2"
+                SFLAGS="${SFLAGS} -DX86_NOCHECK_SSE2"
```

Comment:
Bad idea... See above.

Reply:
the flag "X86_64" is only covered in these places:

CMakeLists.txt:383:    add_definitions(-DX86_64 -DX86_NOCHECK_SSE2 -DUNALIGNED_OK -DUNROLL_LESS)
configure:772:                CFLAGS="${CFLAGS} -DX86_64 -DX86_NOCHECK_SSE2"
configure:773:                SFLAGS="${SFLAGS} -DX86_64 -DX86_NOCHECK_SSE2"
zlib.h:112:#if defined(ZLIB_COMPAT) && defined(X86_64)

So from my view it's only defined for enabling padding

Reply:
The define has existed long time before the padding was added... Patch for one architecture shouldn't touch code for another architecture. That however doesn't include lines that apply for multiple architectures and only change is adding more tests.